base/iterator.jl

```diff
@@ -63,13 +63,13 @@ function tuple_type_cons{S,T<:Tuple}(::Type{S}, ::Type{T})
     Tuple{S, T.parameters...}
 end
 eltype{I,Z}(::Type{Zip{I,Z}}) = tuple_type_cons(eltype(I), eltype(Z))
-start(z::Zip) = tuple(start(z.a), start(z.z))
-function next(z::Zip, st)
+@inline start(z::Zip) = tuple(start(z.a), start(z.z))
```

Comment:
@JeffBezanson Why is `tuple` used here instead  of the standard syntax? You introduced this in https://github.com/JuliaLang/julia/commit/eaf5a959cb30e64e6b3b5fca525d892320d2fbda.